documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,601 @@
+
+#### Reverse flow key
+
+When configuring a flow_entry, it can be specified whether it is unidirectional or bidirectional. If it is bidirectional, it can be designated as a reverse flow key, allowing for the rapid identification of the corresponding reverse flow. Of course, if a flow entry is initially established as unidirectional, its reverse flow can also be set up later, utilizing these attributes to link them together.
+
+| Attribute name                            | Type               | Description                                 |
+| ----------------------------------------- | ------------------ | ------------------------------------------- |
+| SAI_FLOW_ENTRY_ATTR_IS_BIDIRECTIONAL_FLOW | `bool`             | Indicates if the flow is bidirectional      |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_ENI_MAC       | `sai_mac_t`        | Eni mac addr for the recerse flow           |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_IP_PROTO      | `sai_uint8_t`      | IP protocol number for the reverse flow     |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_IP_ADDR       | `sai_ip_address_t` | Source IP address for the reverse flow      |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_IP_ADDR       | `sai_ip_address_t` | Destination IP address for the reverse flow |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_SRC_PORT      | `sai_uint16_t`     | L4 source port for the reverse flow         |
+| SAI_FLOW_ENTRY_ATTR_REVERSE_DST_PORT      | `sai_uint16_t`     | L4 destination port for the reverse flow    |
+
+#### Flow encap
+
+These are the related attributes of flow encapsulation.
+
+| Attribute name                         | Type                       | Description                                     |
+| -------------------------------------- | -------------------------- | ----------------------------------------------- |
+| SAI_FLOW_ENTRY_ATTR_DEST_VNET_VNI      | `sai_uint32_t`             | Destination VNI                                 |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_SIP       | `sai_uint32_t`             | Source IP address in the underlay network       |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_DIP       | `sai_uint32_t`             | Destination IP address in the underlay network  |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_SMAC      | `sai_mac_t`                | Source MAC address in the underlay network      |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY_DMAC      | `sai_mac_t`                | Destination MAC address in the underlay network |
+| SAI_FLOW_ENTRY_ATTR_DASH_ENCAPSULATION | `sai_dash_encapsulation_t` | Encapsulation method for DASH traffic           |
+
+#### Flow rewrite
+
+These are the related attributes of flow rewrite.
+
+| Attribute name               | Type               | Description                                                  |
+| ---------------------------- | ------------------ | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_IS_IPV6  | `bool`             | Indicates whether the flow is IPv6 (true) or IPv4 (false).   |
```

Comment:
this attribute may exist in p4 code but it should be gone after code generation, because the sai_ip_address_t already contains this info.